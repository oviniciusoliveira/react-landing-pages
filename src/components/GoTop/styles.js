import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    position: fixed;
    z-index: 6;
    bottom: 2rem;
    right: 2rem;

    border-radius: ${theme.borderRadius.small};

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    width: 4rem;
    height: 4rem;
  `}
`;
