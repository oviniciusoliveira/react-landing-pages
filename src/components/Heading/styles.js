import styled, { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, lightColor }) => css`
    color: ${lightColor ? theme.colors.white : theme.colors.primaryColor};
  `}
`;
