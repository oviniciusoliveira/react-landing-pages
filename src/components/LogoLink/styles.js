import styled, { css } from 'styled-components';

export const LogoLink = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    & > img {
      height: 3rem;
    }
  `}
`;
