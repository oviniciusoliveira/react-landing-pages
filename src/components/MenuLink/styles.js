import styled, { css } from 'styled-components';

export const MenuLink = styled.a.attrs((props) => ({
  target: props.newTab ? '_blank' : '_self',
}))`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.78rem;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      width: 0;
      left: 50%;
      transition: all 0.2s ease-in-out;
    }

    &:hover::after {
      width: 50%;
      left: 25%;
    }
  `}
`;
