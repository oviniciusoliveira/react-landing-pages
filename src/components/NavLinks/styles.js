import styled, { css } from 'styled-components';

export const NavLinks = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-wrap: wrap;

    @media ${theme.media.ltMedium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`;
