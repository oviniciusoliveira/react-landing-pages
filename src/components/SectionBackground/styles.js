import styled, { css } from 'styled-components';

const lightBackground = (theme) => css`
  background: ${theme.colors.lightBackground};
  color: ${theme.colors.primaryColor};
`;

const darkBackground = (theme) => css`
  background: ${theme.colors.darkBackground};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, withBackground }) => css`
    ${withBackground ? darkBackground(theme) : lightBackground(theme)}
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
