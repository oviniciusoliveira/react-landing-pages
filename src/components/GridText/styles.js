import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Paragraph as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${HeadingContainer} {
    position: relative;
    width: max-content;
    left: 5rem;
  }

  ${HeadingContainer}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -2.5rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`;
