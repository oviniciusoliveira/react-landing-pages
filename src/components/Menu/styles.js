import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as LogoTitle } from '../Heading/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
  height: 100%;
`;

export const Container = styled.div`
  ${({ theme, isVisible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    width: 100%;
    border-bottom: ${theme.colors.mediumGray};
    background-color: ${theme.colors.lightBackground};
    transition: ${isVisible ? 'all 0.2s ease-in-out' : 'none'};

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${LogoTitle} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.ltMedium} {
      visibility: hidden;
      opacity: 0;
      ${isVisible && menuVisible()}

      & ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: scroll;
      }

      & ${LogoTitle} {
        padding-bottom: ${theme.spacings.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.ltMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacings.xxlarge} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    border-radius: 0.3rem;
    display: none;
    cursor: pointer;

    & > svg {
      width: 75%;
      height: 75%;
    }

    @media ${theme.media.ltMedium} {
      display: block;
    }
  `}
`;
