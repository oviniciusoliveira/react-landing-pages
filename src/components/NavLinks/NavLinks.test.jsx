import { screen } from '@testing-library/dom';
import { renderTheme } from '../../test/render-theme';
import { NavLinks } from '.';
import linksMock from './mock';
import { theme } from '../../styles/theme';

describe('<NavLinks />', () => {
  test('should render a nav with links', () => {
    renderTheme(<NavLinks links={linksMock} />);
    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(links.length);
  });

  test('should not render any link', () => {
    renderTheme(<NavLinks />);
    const links = screen.queryAllByRole(/link/i);
    expect(links).toHaveLength(0);
  });

  test('should apply media query when screen size is less than 768px', () => {
    renderTheme(<NavLinks links={linksMock} />);
    const navLinks = screen.getByText(/link 10/i).parentElement;
    expect(navLinks).toHaveStyleRule('flex-flow', 'column nowrap', {
      media: theme.media.ltMedium,
    });
  });

  test('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={linksMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
