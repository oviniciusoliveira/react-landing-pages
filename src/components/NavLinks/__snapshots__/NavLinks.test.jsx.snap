// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  display: block;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.78rem;
  height: 0.2rem;
  background: #dc143c;
  width: 0;
  left: 50%;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c1:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="#target1"
    target="_self"
  >
    Link 1
  </a>
  <a
    class="c1"
    href="#target2"
    target="_self"
  >
    Link 2
  </a>
  <a
    class="c1"
    href="#target3"
    target="_self"
  >
    Link 3
  </a>
  <a
    class="c1"
    href="#target4"
    target="_self"
  >
    Link 4
  </a>
  <a
    class="c1"
    href="#target5"
    target="_self"
  >
    Link 5
  </a>
  <a
    class="c1"
    href="#target6"
    target="_self"
  >
    Link 6
  </a>
  <a
    class="c1"
    href="#target7"
    target="_self"
  >
    Link 7
  </a>
  <a
    class="c1"
    href="#target8"
    target="_self"
  >
    Link 8
  </a>
  <a
    class="c1"
    href="#target9"
    target="_self"
  >
    Link 9
  </a>
  <a
    class="c1"
    href="#target10"
    target="_self"
  >
    Link 10
  </a>
</nav>
`;
